File: firebase/functions/lib/community/get_community_pre_post_enabled.dart
import 'dart:async';

import 'package:firebase_functions_interop/firebase_functions_interop.dart';
import '../on_call_function.dart';
import '../utils/subscription_plan_util.dart';
import 'package:data_models/cloud_functions/requests.dart';

class GetCommunityPrePostEnabled
    extends OnCallMethod<GetCommunityPrePostEnabledRequest> {
  GetCommunityPrePostEnabled()
      : super(
          'GetCommunityPrePostEnabled',
          (jsonMap) => GetCommunityPrePostEnabledRequest.fromJson(jsonMap),
        );

  @override
  Future<Map<String, dynamic>> action(
    GetCommunityPrePostEnabledRequest request,
    CallableContext context,
  ) async {
    final capabilities =
        await subscriptionPlanUtil.calculateCapabilities(request.communityId);
    return GetCommunityPrePostEnabledResponse(
      prePostEnabled: capabilities.hasPrePost!,
    ).toJson();
  }
}

File: firebase/functions/src/community/get_community_pre_post_enabled.ts
import * as functions from 'firebase-functions'
import { OnCallMethod } from '../on_call_function'
import { GetCommunityPrePostEnabledRequest } from '../types'
import { subscriptionPlanUtil } from '../utils/subscription_plan_util'

export class GetCommunityPrePostEnabled extends OnCallMethod<GetCommunityPrePostEnabledRequest> {
    constructor() {
        super('GetCommunityPrePostEnabled', (json) => json as GetCommunityPrePostEnabledRequest)
    }

    async action(
        request: GetCommunityPrePostEnabledRequest,
        _context: functions.https.CallableContext
    ): Promise<{ prePostEnabled: boolean }> {
        const capabilities = await subscriptionPlanUtil.calculateCapabilities(request.communityId)
        return { prePostEnabled: capabilities.hasPrePost ?? false }
    }
}
